lib/buffer.js

```diff
@@ -170,18 +170,31 @@ Object.defineProperty(Buffer, Symbol.species, {
  * Buffer.from(arrayBuffer[, byteOffset[, length]])
  **/
 Buffer.from = function(value, encodingOrOffset, length) {
+  if (value === undefined || value === null)
+    throw new TypeError(kFromErrorMsg);
+
+  if (typeof value === 'number')
+    throw new TypeError('"value" argument must not be a number');
+
   if (typeof value === 'string')
```

Comment:
IMHO we should be promoting this and the other more common cases to the top of this function. This is why `typeof value === 'number'` was at the very end for example.